{
  "hash": "40c36e2741d7fe1e66d6ec55477ba5fd19c9c5d2",
  "hash_short": "40c36e27",
  "subject": "x86/mce: Fix incorrect \"Machine check from unknown source\" message",
  "body": "Some injection testing resulted in the following console log:\n\n  mce: [Hardware Error]: CPU 22: Machine Check Exception: f Bank 1: bd80000000100134\n  mce: [Hardware Error]: RIP 10:<ffffffffc05292dd> {pmem_do_bvec+0x11d/0x330 [nd_pmem]}\n  mce: [Hardware Error]: TSC c51a63035d52 ADDR 3234bc4000 MISC 88\n  mce: [Hardware Error]: PROCESSOR 0:50654 TIME 1526502199 SOCKET 0 APIC 38 microcode 2000043\n  mce: [Hardware Error]: Run the above through 'mcelog --ascii'\n  Kernel panic - not syncing: Machine check from unknown source\n\nThis confused everybody because the first line quite clearly shows\nthat we found a logged error in \"Bank 1\", while the last line says\n\"unknown source\".\n\nThe problem is that the Linux code doesn't do the right thing\nfor a local machine check that results in a fatal error.\n\nIt turns out that we know very early in the handler whether the\nmachine check is fatal. The call to mce_no_way_out() has checked\nall the banks for the CPU that took the local machine check. If\nit says we must crash, we can do so right away with the right\nmessages.\n\nWe do scan all the banks again. This means that we might initially\nnot see a problem, but during the second scan find something fatal.\nIf this happens we print a slightly different message (so I can\nsee if it actually every happens).\n\n[ bp: Remove unneeded severity assignment. ]\n\nSigned-off-by: Tony Luck <tony.luck@intel.com>\nSigned-off-by: Borislav Petkov <bp@suse.de>\nSigned-off-by: Thomas Gleixner <tglx@linutronix.de>\nCc: Ashok Raj <ashok.raj@intel.com>\nCc: Dan Williams <dan.j.williams@intel.com>\nCc: Qiuxu Zhuo <qiuxu.zhuo@intel.com>\nCc: linux-edac <linux-edac@vger.kernel.org>\nCc: stable@vger.kernel.org # 4.2\nLink: http://lkml.kernel.org/r/52e049a497e86fd0b71c529651def8871c804df0.1527283897.git.tony.luck@intel.com",
  "full_message": "x86/mce: Fix incorrect \"Machine check from unknown source\" message\n\nSome injection testing resulted in the following console log:\n\n  mce: [Hardware Error]: CPU 22: Machine Check Exception: f Bank 1: bd80000000100134\n  mce: [Hardware Error]: RIP 10:<ffffffffc05292dd> {pmem_do_bvec+0x11d/0x330 [nd_pmem]}\n  mce: [Hardware Error]: TSC c51a63035d52 ADDR 3234bc4000 MISC 88\n  mce: [Hardware Error]: PROCESSOR 0:50654 TIME 1526502199 SOCKET 0 APIC 38 microcode 2000043\n  mce: [Hardware Error]: Run the above through 'mcelog --ascii'\n  Kernel panic - not syncing: Machine check from unknown source\n\nThis confused everybody because the first line quite clearly shows\nthat we found a logged error in \"Bank 1\", while the last line says\n\"unknown source\".\n\nThe problem is that the Linux code doesn't do the right thing\nfor a local machine check that results in a fatal error.\n\nIt turns out that we know very early in the handler whether the\nmachine check is fatal. The call to mce_no_way_out() has checked\nall the banks for the CPU that took the local machine check. If\nit says we must crash, we can do so right away with the right\nmessages.\n\nWe do scan all the banks again. This means that we might initially\nnot see a problem, but during the second scan find something fatal.\nIf this happens we print a slightly different message (so I can\nsee if it actually every happens).\n\n[ bp: Remove unneeded severity assignment. ]\n\nSigned-off-by: Tony Luck <tony.luck@intel.com>\nSigned-off-by: Borislav Petkov <bp@suse.de>\nSigned-off-by: Thomas Gleixner <tglx@linutronix.de>\nCc: Ashok Raj <ashok.raj@intel.com>\nCc: Dan Williams <dan.j.williams@intel.com>\nCc: Qiuxu Zhuo <qiuxu.zhuo@intel.com>\nCc: linux-edac <linux-edac@vger.kernel.org>\nCc: stable@vger.kernel.org # 4.2\nLink: http://lkml.kernel.org/r/52e049a497e86fd0b71c529651def8871c804df0.1527283897.git.tony.luck@intel.com",
  "author_name": "Tony Luck",
  "author_email": "tony.luck@intel.com",
  "author_date": "Fri Jun 22 11:54:23 2018 +0200",
  "author_date_iso": "2018-06-22T11:54:23+02:00",
  "committer_name": "Thomas Gleixner",
  "committer_email": "tglx@linutronix.de",
  "committer_date": "Fri Jun 22 14:35:50 2018 +0200",
  "committer_date_iso": "2018-06-22T14:35:50+02:00",
  "files_changed": [
    "arch/x86/kernel/cpu/mcheck/mce.c"
  ],
  "files_changed_count": 1,
  "stats": [
    {
      "file": "arch/x86/kernel/cpu/mcheck/mce.c",
      "insertions": 18,
      "deletions": 8
    }
  ],
  "total_insertions": 18,
  "total_deletions": 8,
  "total_changes": 26,
  "parents": [
    "1f74c8a64798e2c488f86efc97e308b85fb7d7aa"
  ],
  "branches": [
    "* development",
    "master",
    "remotes/origin/HEAD -> origin/master",
    "remotes/origin/master"
  ],
  "tags": [
    "v4.18",
    "v4.18-rc2",
    "v4.18-rc3",
    "v4.18-rc4",
    "v4.18-rc5",
    "v4.18-rc6",
    "v4.18-rc7",
    "v4.18-rc8",
    "v4.19",
    "v4.19-rc1"
  ],
  "is_merge": false,
  "security_info": {
    "cve_ids": [],
    "security_keywords": [
      "injection"
    ]
  },
  "fix_type": "security",
  "file_results": [
    {
      "file": "arch/x86/kernel/cpu/mcheck/mce.c",
      "pre_version": true,
      "post_version": true,
      "patch": true
    }
  ]
}